### PR TITLE
Add custom quad meshing factor to PCCMG

### LINKED ISSUE #21030: Make PCCMG Central Quad Element Meshing More Flexible
## Reason
`PolygonConcentricCircleMeshGenerator` is capable of performing all quad element meshing. In the center region, current algorithm uses a fixed factor N/(N+1) as shown in the following image in the [class documentation page](https://mooseframework.inl.gov/source/meshgenerators/PolygonConcentricCircleMeshGenerator.html).
quad_cent
Based on some users' feedback, it would be helpful to make this factor flexible so that users can modify it in input files.

## Design
Add an input parameter to control the aforementioned factor.

## Impact
Make `PolygonConcentricCircleMeshGenerator` more flexible for users.

### PR DESCRIPTION
**Pull request by miaoyinb**
closes #21030

### CODE REVIEW

**Comment by miaoyinb** [comment]
@loganharbour @GiudGiud This is a minor update to allow users to adjust a parameter used in PCCMG.

@aprilnovak FYI

File: modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C [hunk 1]
```diff
@@ -277,6 +280,8 @@ PolygonMeshGeneratorBase::centerNodes(ReplicatedMesh & mesh,
   const std::pair<Real, Real> p_origin = std::make_pair(0.0, 0.0);
   const std::pair<Real, Real> p_bottom =
       std::make_pair(0.0, ring_radii_0 * std::cos(M_PI / side_number));
+  // const std::pair<Real, Real> p_bottom =
+  //     std::make_pair(0.0, ring_radii_0);
```
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C:284]: remove

File: modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C [hunk 2]
```diff
@@ -167,6 +167,10 @@ PolygonConcentricCircleMeshGeneratorBase::validParams()
                         "Whether the side elements are reorganized to have a uniform size.");
   params.addParam<bool>(
       "quad_center_elements", false, "Whether the center elements are quad or triangular.");
+  params.addRangeCheckedParam<Real>(
+      "center_quad_factor",
+      "center_quad_factor>0&center_quad_factor<1",
+      "A factor used to support customized quad center nodes determination.");
```
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C:173]: needs to be more descriptive.
It's a ratio of distances right

Comment (GiudGiud) [reply modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C:173]: let's adjust the 3 docstrings with the more descriptive too please

**Review by GiudGiud — CHANGES_REQUESTED** [review]

File: modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C (hunk 2 above)
Comment (miaoyinb) [reply modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C:173]: Yes, may be call it "a factor used to determine the locations of transition nodes in the center region meshed by quad elements"?

Comment (GiudGiud) [reply modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C:173]: it still doesnt explicitly say how it determines this. 

btw would it fit in one of the parameter groups?

Comment (GiudGiud) [reply modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C]: seems good to me.

So the range check bans 0 so this part is confusing. Users will try to set a 0 to get the default behavior. 
I would rather we dont talk about this 0, just say: default value is ....

**Comment by miaoyinb** [comment]
@bwspenc @dschwen @jiangwen84 The review requests were sent by mistake (messed up something during a rebase). Sorry.

**Review by GiudGiud — APPROVED** [review]

**Comment by miaoyinb** [comment]
Thanks @GiudGiud

**Comment by GiudGiud** [comment]
Thanks for the contribution! Cant wait to have these meshes on the vtb

**Pull request merged**